polling/src/main/java/io/atleon/polling/reactive/Poller.java

```diff
@@ -0,0 +1,34 @@
+package io.atleon.polling.reactive;
+
+import java.util.Collection;
+
+/**
+ * The Poller class is what manages the interaction with the external resource to be polled. The Poller
+ * should be implemented by the user and supplied to {@link PollingReceiver} when instantiated. The supplied
+ * implementation will need to have an internal mechanism to track offsets into the external resource as this
+ * poll() method has no parameters and is only expecting to receive the next batch of events for the stream.
+ * @param <K> The type of the items returned by the external resource (Must implement {@link PollingRecord}.
+ * @param <V> The data type of the data needed for Acking and Nacking.
+ */
+public interface Poller<K extends PollingRecord<V>, V> {
+
+    /**
+     * This method will be called periodically to fetch new records to push into the stream.
+     * @return - A collection of records from the external resource.
+     */
+    Collection<K> poll();
```

Comment:
I have an idea that might make this interface simpler: You're already having to define a `PollingRecord` (think this might make more sense to call `PolledRecord`, since it will be the emission resulting from polling). I think that interface inherently has three behaviors associated with it:
1. Indicating what native "group" it belongs to. In the case of Kafka, this is the topic-partition the record came from. In the case of RabbitMQ, there is no grouping, so the group would always return some constant (maybe `Optional.empty()`). As another example, Kinesis records all belong to a shard. For time-based polling, there might be no group, so would return `Optional.empty()`
2. Providing the information necessary to ack this message back to the Pollable resource. I've got the current name of "marker" stuck in my head for this. In the case of Kafka, this would be the topic-partition-offset. In the case of RabbitMQ, this would be the delivery tag. I think a similar concept would apply for time-based polling
3. Providing the "payload" representing the data that was polled. In the case of Kafka, this would be the ConsumerRecord. In the case of RabbitMQ, this would be the Delivery. etc...


Proposal: Define a `PolledRecord<P>` interface like the following:
```java
public interface PolledRecord<M, V> {

    default Optional<?> nativeGroup() {
        return Optional.empty();
    }

    M marker();

    V payload();
}
```

Then the Poll implementation just needs to be generic in terms of the marker and the payload, and look like this:
```java
public interface Pollable<M, R extends PolledRecord<M, ?>> {

    Collection<R> poll();

    void ack(M marker);

    void nack(M marker, Throwable error);
}
```

WDYT?

Reply:
Yeah, I like that, what do you think about actually making `PolledRecord` a class with the same contract, that way the implementation doesn't have to include an implementation and can instead just wrap its domain model?